lib/membrane_mp4/muxer/cmaf/header.ex

```diff
@@ -4,22 +4,10 @@ defmodule Membrane.MP4.Muxer.CMAF.Header do
 
   @ftyp FileTypeBox.assemble("iso5", ["iso6", "mp41"])
 
-  @spec serialize(%{
-          timescale: integer,
-          width: :integer,
-          height: :integer,
-          content: struct
-        }) :: binary
-  def serialize(config) do
-    track =
-      config
-      |> Map.take([:timescale, :width, :height, :content])
-      |> Map.put(:id, 1)
-      |> Track.new()
-      |> List.wrap()
-
-    movie_extends = MovieBox.MovieExtendsBox.assemble(track)
-    movie_box = MovieBox.assemble(track, movie_extends)
+  @spec serialize([Track.t()]) :: binary
+  def serialize(tracks) do
```

Comment:
On the other hand, when using this function you refer to it as `Header.serialize(tracks)`, so the module kind of suggests serializing a header. In other modules representing MP4 boxes (e.g. `MovieBox`) we have `assemble` functions, some of them are also taking one or many tracks, but are assembling boxes, not tracks 😛 . Also, `serialize` clearly indicates that we are getting a binary, when `generate` sounds more ambiguous.
I don't mean `generate` would not fit here, I'm just not sure if it would be beneficial in the end.

Reply:
Actually, I can see why @mickel8 doesn't like `Header.serialize/1` for the function that both assembles and serializes the header. However, `Header.serialize` is consistent with the rest of the plugin, so if you think this needs some adjustment, it would probably be better to leave an issue about it and sort it out at a later date